File: src/main/scala/net/machinemuse/numina/sound/MovingSoundPlayer.scala
package net.machinemuse.numina.sound

import net.minecraft.client.audio.ISound.AttenuationType
import net.minecraft.client.audio.MovingSound
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.{ResourceLocation, SoundCategory, SoundEvent}

/**
 * Created by Claire on 8/27/2015.
 */

// TODO: fix this the proper way. The last 2 params were added to fix building, but it won't be compatible with MPS
class MovingSoundPlayer(val player:EntityPlayer,
                        soundIn: SoundEvent,
                        newvolume:Float,
                        newpitch:Float,
                        newrepeat:Boolean = false,
                        categoryIn: SoundCategory) extends MovingSound(soundIn, categoryIn) {

  this.pitch = newpitch
  this.volume = newvolume
  this.repeat = newrepeat

  def updatePitch (newpitch:Float) = {
    pitch = newpitch
    this
  }

  def updateVolume (newvolume:Float) = {
    volume = (4.0f*volume + newvolume)/5.0f
    this
  }

  def updateRepeat (newrepeat:Boolean) = {
    repeat = newrepeat
    this
  }



  override def getAttenuationType: AttenuationType = AttenuationType.LINEAR

  def stopPlaying() = {
    this.donePlaying = true
  }

  override def update(): Unit = {
    this.xPosF = player.posX.toFloat
    this.yPosF = player.posY.toFloat
    this.zPosF = player.posZ.toFloat
  }

}

File: src/main/java/net/machinemuse/numina/sound/MovingSoundPlayer.java
package net.machinemuse.numina.sound;

import net.minecraft.client.audio.ISound;
import net.minecraft.client.audio.MovingSound;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.ResourceLocation;

/**
 * Ported to Java by lehjr on 10/22/16.
 */
public class MovingSoundPlayer extends MovingSound {
    private static EntityPlayer player;


    public MovingSoundPlayer(final EntityPlayer player, final ResourceLocation resourceLocation, final float newvolume, final float newpitch, final boolean newrepeat) {
        super(resourceLocation);
        this.player = player;

        this.field_147663_c = newpitch;
        this.volume = newvolume;
        this.repeat = newrepeat;
    }

    public EntityPlayer player() {
        return this.player;
    }

    public MovingSoundPlayer updatePitch(final float newpitch) {
        this.field_147663_c = newpitch;
        return this;
    }

    public MovingSoundPlayer updateVolume(final float newvolume) {
        this.volume = (4.0f * this.volume + newvolume) / 5.0f;
        return this;
    }

    public MovingSoundPlayer updateRepeat(final boolean newrepeat) {
        this.repeat = newrepeat;
        return this;
    }

    @Override
    public ISound.AttenuationType getAttenuationType() {
        return ISound.AttenuationType.LINEAR;
    }

    public void stopPlaying() {
        super.donePlaying = true;
    }

    @Override
    public void update() {
        this.xPosF = (float)this.player().posX;
        this.yPosF = (float)this.player().posY;
        this.zPosF = (float)this.player().posZ;
    }
}